src/dbnode/persist/fs/read.go

```diff
@@ -382,40 +386,28 @@ func (r *reader) readInIndexedOrder() (ident.ID, ident.TagIterator, checked.Byte
 		return nil, nil, nil, 0, err
 	}
 
-	var data checked.Bytes
-	if r.bytesPool != nil {
-		data = r.bytesPool.Get(int(entry.Size))
-		data.IncRef()
-		defer data.DecRef()
-	} else {
-		data = checked.NewBytes(make([]byte, 0, entry.Size), nil)
-		data.IncRef()
-		defer data.DecRef()
-	}
-
-	if entry.Offset+entry.Size > int64(len(r.dataMmap.Bytes)) {
-		return nil, nil, nil, 0, fmt.Errorf(
-			"attempt to read beyond data file size (offset=%d, size=%d, file size=%d)",
-			entry.Offset, entry.Size, len(r.dataMmap.Bytes))
-	}
-
-	data.AppendAll(r.dataMmap.Bytes[entry.Offset : entry.Offset+entry.Size])
+	data := r.dataMmap.Bytes[entry.Offset : entry.Offset+entry.Size]
```

Comment:
Hm, I thought we had put in protection to see if entry.Offset+entry.Size isn't out of bands here?  Looks like we don't have any here, should we add it?

Ah I see we removed it, can we add it back in? (that is `if entry.Offset+entry.Size > int64(len(r.dataMmap.Bytes)) {`)

Reply:
My bad, looks like this was lost during merge conflict resolving.
Will fix this, and also go through the comments from the last PR on these files, to make sure nothing else is missing.

Reply:
Resurrected the bounds check on mmaped bytes, also checked the status of other comments after the merge, didn't notice anything else that would be missing.